app/src/pages/index.js

```diff
@@ -5,16 +5,38 @@ import SEO from "../components/seo"
 import BookCard from "../components/bookcard"
 
 export default ({data}) => {
+  let max=data.allBooksJson.edges.length, init=4, end=8;
+  function hide(){
+    for(var i=init; i<document.getElementsByClassName("book").length; i++){
+      document.getElementsByClassName("book")[i].hidden=true;
+    }
+  }
+  window.document.onscroll = () => myFunction();
+  function myFunction() {
+    if (document.documentElement.clientHeight+document.documentElement.scrollTop===document.documentElement.scrollHeight && end<=max) {
+      for(var i=init; i<end; i++){
```

Comment:
Instead of doing this how about just fetching the next N books greater than ID of last book we appended to dom?